### PR TITLE
Bull remastered: The rebullening: The bull within

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Removing Bull is lame.

Bull can now change direction while charging without primo.
Changing direction mid charge now has a cooldown, so you can change direction, but you can't do so repeatedly in a short period.
 Primo reduces this cool down to make you more agile, so is now far less overwhelming.
Note: You cannot double back on yourself without losing charge, so no more going back and forth to endlessly charge a poor unga.

Primo bull now also gets the stomp ability, since it's going to be notably weaker than before. It also means it can do something other that literally charge and nothing else.

## Why It's Good For The Game
Primobull is bullshit, non primo is bullbad
Addresses both of these points somewhat

## Changelog
:cl:
add: Bulls can now natively change direction when charging except for 180 degrees, on a cooldown
add: Primo bull reduces charge direction change cooldown and has the stomp ability
/:cl:

### CODE REVIEW

**Comment by Wayland-Smithy** [comment]
Cool, something other to do than WASD with stomp and I agree the 180 degree turns are BS. Seems fair.

My only fear is the lack of feedback as to when you are able to turn again and keep your speed will lead to situations where you screw yourself. Seems like it could be a noob trap, and might not be great to need to "learn the feel" for how long since your last turn especially when time dilation exists.

**Comment by Lumipharon** [comment]
oh wait, this will effect crusher as well, aylmao. I'll update it.

**Comment by Wayland-Smithy** [comment]
Lol keep it!

**Comment by Mrrpip** [comment]
Deja vu crusher time?

**Comment by Al-1ce** [comment]
bring back panther next I double-dog dare you

**Comment by FosterGrime-Heymynameslouie** [comment]
I dont excatly know how this will impact bull gameplay ™ and if it would still be able to chain stun an entire push but im thumbs upping none theless because of the pr title

**Comment by iLikeMemes** [comment]
I just had a (possibly horrendous) Ideaguys moment:

What if bull had a charge that pins marines to the bull and makes them stick to the bull for a certain amount of tiles? Could Make for funny displacement opportunities.

I dont know if this would actually fit bull but since you are doing some funky changes to bull i might as well randomly throw this out there.

Kind of loosely got inspiration from left 4 dead charger.

**Comment by Losenis** [comment]
>Bull can now change direction while charging without primo.

This is genius. This will have everyone notice how cancer it is without requiring primo to be bought so more people will support its removal without seeing it as a primo-only thing.

**Comment by Wayland-Smithy** [comment]
> This is genius. This will have everyone notice how cancer it is without requiring primo to be bought so more people will support its removal without seeing it as a primo-only thing.

You may not realize how much of a nerf this is by allowing 1 direction change every 1.6 seconds and even at primo being half it is still going to mean keeping a charge very difficult. No more will a 1 or 2 wide lane be a death trap for ungas when it takes over 3 seconds to turn around.

**Comment by Lumipharon** [comment]
> > Bull can now change direction while charging without primo.
> 
> This is genius. This will have everyone notice how cancer it is without requiring primo to be bought so more people will support its removal without seeing it as a primo-only thing.

Yeah this severely cucks primo bull compared to how it is now. You'll need very large open areas to actuall be able to double back without ending your charge, but it does mean you can charge around corners.

Some new fancy ability that varies on your charge stance would be cool but uh, that would involve thinking so cbf right now.

**Comment by TiviPlus** [comment]
The new primo is extremely lazy

**Comment by Wayland-Smithy** [comment]
Would love to see primo keep the ability to change dir as fast as possible now that the 180 degree double back method will no longer work which is **the most cheesy aspect** of how primo is now. A lower damage stomp always being available would be cool too. This would really cement the difference between Crusher as a lane needing defense smashing tanky beast, and Bull as the agile light weight relative aspiring to get swole.

I find it very enjoyable to strive towards the skill ceiling of maintaing a charge through a long convoluted lane-less maze which would no longer be possible at primo if the changes stand as is.

File: code/modules/mob/living/carbon/xenomorph/charge_crush.dm [hunk 1]
```diff
@@ -34,8 +34,10 @@
 	var/crush_living_damage = 20
 	var/next_special_attack = 0 //Little var to keep track on special attack timers.
 	var/plasma_use_multiplier = 1
-	///If this charge should keep momentum on dir change and if it can charge diagonally
-	var/agile_charge = FALSE
+	///If this charge should keep momentum on dir change
+	var/agile_charge = 0
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/charge_crush.dm:38]: Comment no longer fits var

File: code/modules/mob/living/carbon/xenomorph/charge_crush.dm [hunk 2]
```diff
@@ -89,10 +91,17 @@
 	var/mob/living/carbon/xenomorph/charger = owner
 	if(charger.is_charging == CHARGE_OFF)
 		return
-	if(!old_dir || !new_dir || old_dir == new_dir || agile_charge) //Check for null direction from help shuffle signals
+	if(!old_dir || !new_dir || old_dir == new_dir) //Check for null direction from help shuffle signals
 		return
-	do_stop_momentum()
-
+	if((!agile_charge) || new_dir == REVERSE_DIR(old_dir)) //No doubling back
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/charge_crush.dm:96]: Do not treat agile charge like a bool, it is an int now

File: code/modules/mob/living/carbon/xenomorph/charge_crush.dm [hunk 3]
```diff
@@ -89,10 +91,17 @@
...
+		do_stop_momentum()
+		return
+	var/turn_rate = 16
+	if(agile_charge == 2)
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/charge_crush.dm:100]: 2 should be a define

File: code/modules/mob/living/carbon/xenomorph/charge_crush.dm [hunk 4]
```diff
@@ -89,10 +91,17 @@
...
 		return
-	if(!old_dir || !new_dir || old_dir == new_dir || agile_charge) //Check for null direction from help shuffle signals
+	if(!old_dir || !new_dir || old_dir == new_dir) //Check for null direction from help shuffle signals
 		return
-	do_stop_momentum()
-
+	if((!agile_charge) || new_dir == REVERSE_DIR(old_dir)) //No doubling back
+		do_stop_momentum()
+		return
+	var/turn_rate = 16
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/charge_crush.dm:99]: Uneeded var, use trinary

File: code/modules/mob/living/carbon/xenomorph/charge_crush.dm [hunk 5]
```diff
@@ -334,6 +342,7 @@
 	max_steps_buildup = 10
 	crush_living_damage = 15
 	plasma_use_multiplier = 2
+	agile_charge = 1
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/charge_crush.dm:345]: This should be a define

File: code/modules/mob/living/carbon/xenomorph/charge_crush.dm [hunk 6]
```diff
@@ -370,7 +379,8 @@
 
 /datum/action/xeno_action/ready_charge/bull_charge/on_xeno_upgrade()
 	var/mob/living/carbon/xenomorph/X = owner
-	agile_charge = (X.upgrade == XENO_UPGRADE_FOUR)
+	if(X.upgrade == XENO_UPGRADE_FOUR)
+		agile_charge = 2
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/charge_crush.dm:383]: What about going from primo to ancient, like you can do it in testing or admin.

And the previous form was better, should be a trinary

**Review by BraveMole — CHANGES_REQUESTED** [review]
Agreed with tivi, the primo is lazy.

Bull being agile by default is fine

I'd say rather than slowing down suddendly if you cross a threshold, just make it slow down on each turn

**Comment by DemonOfHatred** [comment]
Yeah after seeing mechs literally just eat a million slashes and be fine this change really isnt needed.

**Comment by Carnage068** [comment]
I fucking love Primo Bull but this is honestly probably for the best. Going back and forth in a one tile corridor and ramming someone 18 times was fun as hell, but it didn't really make too much sense, LOL. Diagonal charge also made it like, pretty much impossible NOT to be charging all the time as Primo.
Not sure how to feel about the cooldown on changing directions- Seems like it lowers the skill ceiling a lot while kind of taking away part of what makes Bull different from Crusher.

**Pull request closed without merging**